### PR TITLE
Renaming WebGPU to WebGPU-SPIRV (ala Metal-SPIRV).

### PR DESCRIPTION
**Pull request by benvanik**
This is more consistent and lets us separate the device (WebGPU) from the compilation path that targets it (SPIR-V to WGSL). This also sets us up for another path to lowering to WGSL in the future (same reason we have Metal-SPIRV).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -385,7 +385,7 @@ cmake_dependent_option(IREE_TARGET_BACKEND_CUDA "Enables the 'cuda' compiler tar
 # Non-default target backends either have additional dependencies or are
 # experimental/niche in some fashion.
 # Disable WebGPU by default - it has complex deps and is under development.
-cmake_dependent_option(IREE_TARGET_BACKEND_WEBGPU "Enables the 'webgpu' compiler target backend" OFF ${IREE_BUILD_COMPILER} OFF)
+cmake_dependent_option(IREE_TARGET_BACKEND_WEBGPU_SPIRV "Enables the 'webgpu' compiler target backend" OFF ${IREE_BUILD_COMPILER} OFF)
```
Comment (ScottTodd) [inline CMakeLists.txt:388]: We could forward the old option to the new option if we wanted to ease the transition. Probably fine to just push forward though. I know I'll have a few local configs to update, but I don't expect others to really be impacted.

Here's some (verbose) forwarding we used before: https://github.com/openxla/iree/blob/adeb538647f1b8b588fd4d1d115df4f8ed3e0fd7/CMakeLists.txt#L86-L94

File: build_tools/cmake/presets/options.json
Comment (ScottTodd) [inline build_tools/cmake/presets/options.json:1]: Did you intend to format this file? Could 'save without formatting' to still make your replacement(s) in it. (Or keep the formatting changes, doesn't matter too much to me)

File: compiler/plugins/target/WebGPUSPIRV/SPIRVToWGSL.h
Comment (ScottTodd) [inline compiler/plugins/target/WebGPUSPIRV/SPIRVToWGSL.h:1]: Could update the header guard in this file `IREE_COMPILER_PLUGINS_TARGET_WEBGPU_SPIRVTOWGSL_H_` --> `IREE_COMPILER_PLUGINS_TARGET_WEBGPUSPIRV_SPIRVTOWGSL_H_`

matching https://github.com/openxla/iree/blob/adeb538647f1b8b588fd4d1d115df4f8ed3e0fd7/compiler/plugins/target/MetalSPIRV/SPIRVToMSL.h#L7-L8

File: CMakeLists.txt (hunk 1 above)
Comment (benvanik) [reply CMakeLists.txt:388]: good idea - if literally anyone were using this I'd do it, but you may be the only human to compile this today :)

File: compiler/plugins/target/WebGPUSPIRV/SPIRVToWGSL.h
Comment (benvanik) [reply compiler/plugins/target/WebGPUSPIRV/SPIRVToWGSL.h:1]: good catch

File: build_tools/cmake/presets/options.json
Comment (benvanik) [reply build_tools/cmake/presets/options.json:1]: not intentionally, but I'd rather have it on our standard formatting

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**